django_tables2/columns/templatecolumn.py

```diff
@@ -53,24 +53,19 @@ def render(self, record, table, value, bound_column, **kwargs):
         # If the table is being rendered using `render_table`, it hackily
         # attaches the context to the table as a gift to `TemplateColumn`.
         context = getattr(table, "context", Context())
-        context.update(self.extra_context)
-        context.update(
-            {
-                "default": bound_column.default,
-                "column": bound_column,
-                "record": record,
-                "value": value,
-                "row_counter": kwargs["bound_row"].row_counter,
-            }
-        )
-
-        try:
+        additional_context = {
+            "default": bound_column.default,
+            "column": bound_column,
+            "record": record,
+            "value": value,
+            "row_counter": kwargs["bound_row"].row_counter,
+        }
+        additional_context.update(self.extra_context)
+        with context.update(additional_context):
```

Comment:
Did not know about the ability to use `Context.update()` as a context manager, nice!